### PR TITLE
162 theme provider

### PR DESCRIPTION
**Pull request by Thenlie**
Added MUI theme provider. Moved theme logic to app wrapper.

We will still need to fill out the theme and set all the colors we want. Feel free to do this as you develop.

- #162

### CODE REVIEW

**Review by maxh1231 — APPROVED** [review]
Sweet!

File: src/AppWrapper.tsx [hunk 1]
```diff
@@ -29,6 +33,41 @@ export default function AppWrapper(): JSX.Element {
...
+    // On app load, check for browser preferred color scheme
+    useEffect(() => {
+        const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+
+        if (
+            localStorage.theme === 'dark' ||
+            (!('theme' in localStorage) &&
+                window.matchMedia('(prefers-color-scheme: dark)').matches)
+        ) {
+            setTheme(darkTheme);
```
Comment (jadenrapp) [inline src/AppWrapper.tsx:45]:
```suggestion
        ) {
            document.documentElement.classList.add('dark');
            setTheme(darkTheme);
```

File: src/AppWrapper.tsx [hunk 2]
```diff
@@ -29,6 +33,41 @@ export default function AppWrapper(): JSX.Element {
...
+        } else {
+            setTheme(lightTheme);
+        }
```
Comment (jadenrapp) [inline src/AppWrapper.tsx:48]:
```suggestion
        } else {
            document.documentElement.classList.remove('dark');
            setTheme(lightTheme);
        }
```

**Review by jadenrapp — COMMENTED** [review]
Without those lines the app will not set the theme upon first site visit, it will default to light mode.

File: src/AppWrapper.tsx (hunk 1 above)
Comment (Thenlie) [reply src/AppWrapper.tsx]: Nice catch! This has been updated. ✅

File: src/AppWrapper.tsx (hunk 2 above)
Comment (Thenlie) [reply src/AppWrapper.tsx]: Updated. ✅

**Pull request merged**